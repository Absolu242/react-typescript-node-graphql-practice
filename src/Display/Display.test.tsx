import { render, fireEvent } from "@testing-library/react"
import DisplayText from "./Display"
import "@testing-library/jest-dom/extend-expect"

describe("Test DisplayText", () => {
  it("renders without crashing", () => {
    const { baseElement } = render(<DisplayText />)
    console.log(baseElement.innerHTML)
    expect(baseElement).toBeInTheDocument()
  })
  it("receives input text", () => {
    const testuser = "testuser"
    const { getByTestId } = render(<DisplayText />)
    const input = getByTestId("user-input")
    fireEvent.change(input, { target: { value: testuser } })
    expect(input).toBeInTheDocument()
    expect(input).toHaveValue(testuser)
  })
})
