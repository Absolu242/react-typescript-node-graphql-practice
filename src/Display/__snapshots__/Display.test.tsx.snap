// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DisplayText matches snapshot 1`] = `
<body>
  <div>
    <div>
      <form>
        <div>
          <label>
            Enter your name
          </label>
        </div>
        <div>
          <input
            data-testid="user-input"
            type="text"
            value=""
          />
        </div>
        <button
          data-testid="input-submit"
        >
          Show Message
        </button>
        <div>
          <label
            data-testid="final-msg"
          />
        </div>
        <div>
          this is just a test entry
        </div>
      </form>
    </div>
  </div>
</body>
`;
